crates/evm-ee/src/types.rs

```diff
@@ -0,0 +1,348 @@
+//! EVM Execution Environment types.
+//!
+//! This module defines the types needed for EVM block execution within the
+//! ExecutionEnvironment trait framework.
+
+use alloy_consensus::Header;
+use reth_primitives::TransactionSigned;
+use reth_trie::HashedPostState;
+use revm::state::Bytecode;
+use rsp_mpt::EthereumState;
+use strata_codec::{Codec, CodecError};
+use strata_ee_acct_types::{EnvResult, ExecBlock, ExecBlockBody, ExecHeader, ExecPartialState};
+
+pub(crate) type Hash = [u8; 32];
+
+/// Partial state for EVM execution using RSP's sparse Merkle Patricia Trie.
+///
+/// This represents a sparse state containing only the accounts and storage slots
+/// that were accessed during execution. It's optimized for zero-knowledge proof
+/// generation where including the full Ethereum state would be prohibitively expensive.
+///
+/// Also includes witness data needed for block execution:
+/// - Contract bytecodes for executing contract calls
+/// - Ancestor block headers for the BLOCKHASH opcode
+#[derive(Clone, Debug)]
+pub struct EvmPartialState {
+    /// The sparse Merkle Patricia Trie state from RSP
+    ethereum_state: EthereumState,
+    /// Contract bytecodes needed for execution
+    /// FIXME: ensure bytecodes and ancestor_header are right to be placed in this struct
```

Comment:
This is correct, they make sense to be included here so we can look them up.  However, this indexing forces us to reindex it each time.  Surely we can make this a `BTreeMap<B256, Bytecode>` so that we can do the lookup directly, and use a better container structure for the ancestor headers.

Also, conventionally, FIXMEs shouldn't be doc comments, so they won't appear in docs, just use `//`.